### PR TITLE
feat(android): module builds should fail with aar in lib folder in 9.0.0

### PR DESCRIPTION
**Pull request by jquick-axway**
**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-27758

**Summary:**
- Breaking change when building modules with Titanium 9.0.0
- Build system no longer supports AARs in module's "lib" folder.
- Must delete AAR files and reference them via "build.gradle" file instead.

**Note 1:**
Google's build system will not merge a local AAR library into the built AAR, such as its Java classes. So, while it will successfully build, the app will fail at runtime with a `ClassNotFoundException`. Referencing local AAR files is only viable when building an app project.

**Note 2:**
This is not an issue with modules built before Titanium 9.0.0. This is because old modules are turned into gradle subprojects under the app's build folder and their local AARs will be successfully merged by the app build system.

**Test:**
1. Download a zip of the [titanium-web-dialog](https://github.com/appcelerator-modules/titanium-web-dialog) module project.
2. Open the terminal.
3. `CD` to directory: `./titanium-web-dialog/android`
4. Build and run it via: `appc run -p android`
5. Verify that module successfully builds and launches example app in emulator.
6. Create a `lib` subdirectory under: `./titanium-web-dialog/android`
7. Download [exoplayer-core-2.9.6.aar](https://repo.spring.io/libs-release/com/google/android/exoplayer/exoplayer-core/2.9.6/exoplayer-core-2.9.6.aar) to `lib` folder.
8. Download [dexmaker-mockito-2.19.0.jar](https://repo.spring.io/plugins-release/com/linkedin/dexmaker/dexmaker-mockito/2.19.0/dexmaker-mockito-2.19.0.jar) to `lib` folder.
9. Attempt to build via: `appc run -p android`
10. Verify build fails and below error message can be found in log output.
```
[ERROR] [GRADLE] > AAR files are no longer supported under the "lib" directory. You must reference AAR libraries via a "build.gradle" file's dependencies section instead.
```
11. Delete the `exoplayer-core-2.9.6.aar` file.
12. Build and run via: `appc run -p android`
13. Verify app successfully builds and runs example app in emulator.

### CODE REVIEW

**Comment by sgtcoolguy** [comment]
Note this will need a "backport" to 9_0_X or to be cherry-picked once merged.

File: android/templates/module/generated/build.gradle [hunk 1]
```diff
@@ -120,20 +120,34 @@ android {
 			}
 		}
 	}
-
 	compileOptions {
 		sourceCompatibility JavaVersion.VERSION_1_8
 		targetCompatibility JavaVersion.VERSION_1_8
 	}
+	kotlinOptions {
+		jvmTarget = '1.8'
```
Comment (sgtcoolguy) [inline android/templates/module/generated/build.gradle]: Use the same `JavaVersion.VERSION_1_8` reference? as used in the compile options above?

**Review by sgtcoolguy — APPROVED** [review]

File: android/templates/module/generated/build.gradle (hunk 1 above)
Comment (jquick-axway) [reply android/templates/module/generated/build.gradle]: I just confirmed that `JavaVersion.VERSION_1_8` returns `1.8`.
So, yeah, I'll go ahead and do that.
(The above was just blindly following Google's own examples.)

**Comment by lokeshchdhry** [comment]
FR Passed.

Having an `.aar` file in the `libs` folder errors with :
`ERROR] [GRADLE] > AAR files are no longer supported under the "lib" directory. You must reference AAR libraries via a "build.gradle" file's dependencies section instead`

Studio Ver:       6.0.0.201911251516
SDK Ver:          9.0.0 local build
OS Ver:           10.14.5
Xcode Ver:        Xcode 11.3.1
Appc NPM:         4.2.15
Appc CLI:         7.1.2
Daemon Ver:       1.1.3
Ti CLI Ver:       5.2.2
Alloy Ver:        1.14.4
Node Ver:         12.13.1
NPM Ver:          6.12.1
Java Ver:         11.0.1
Android Emu:  ⇨ google Pixel (Android 10)

**Review by garymathews — APPROVED** [review]
CR: **PASS**

**Pull request merged**